src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -179,5 +185,351 @@ private IActionResult RedirectToLocal(string returnUrl)
                 return Redirect("~/");
             }
         }
+
+        /// <summary>
```

Comment:
@MichaelPetrinolis I suppose you copied that as-is from some IdSrv sample? I don't like it (at all: specially the fact the user account is created on the GET callback without explicit user confirmation).

Instead, consider copying what's used in the AccountController of the default VS templates.

Reply:
@PinpointTownes i modified the account controller based on the VS template. I also modified the logic for an existing user (same email), I force him to to login with the local account in order to link the external login